Adafruit_MQTT.cpp

```diff
@@ -508,10 +508,32 @@ void Adafruit_MQTT::processPackets(int16_t timeout) {
   }
 }
 Adafruit_MQTT_Subscribe *Adafruit_MQTT::readSubscription(int16_t timeout) {
-  // Check if data is available to read.
-  uint16_t len =
-      readFullPacket(buffer, MAXBUFFERSIZE, timeout); // return one full packet
-  return handleSubscriptionPacket(len);
+
+  // Sync or Async subscriber with message
+  Adafruit_MQTT_Subscribe *s = 0;
+
+  // Check if are unread messages
+  for (uint8_t i = 0; i < MAXSUBSCRIPTIONS; i++) {
+    if (subscriptions[i] && subscriptions[i]->new_message) {
+      s = subscriptions[i];
+      break;
+    }
+  }
+
+  // not unread message
+  if (!s) {
+    // Check if data is available to read.
+    uint16_t len = readFullPacket(buffer, MAXBUFFERSIZE,
+                                  timeout); // return one full packet
+    s = handleSubscriptionPacket(len);
```

Comment:
Why are we not marking `new_message` here?

Reply:
new_message is a flag that indicates that the subscriber received a message that has not been read yet. This flag is set to true within handleSubscriptionPacket